lab-christian/route/photo-router.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const AWS = require('aws-sdk');
+const multer = require('multer');
+const Router = require('express').Router;
+const createError = require('http-errors');
+const debug = require('debug')('instaclone:photo-router');
+
+const Photo = require('../model/photo');
+const Gallery = require('../model/gallery');
+const bearerAuth = require('../lib/bearer-auth-middleware');
+
+const photoRouter = module.exports = Router();
+
+AWS.config.setPromisesDependency(require('bluebird'));
+
+const s3 = new AWS.S3();
+const dataDir = `${__dirname}/../data`;
+const upload = multer( { dest: dataDir } );
+
+function s3uploadProm(params) {
+  debug('s3uploadProm');
+  return new Promise((resolve) => {
+    s3.upload(params, (err, s3data) => {
```

Comment:
you can natively use promises with s3 by chaining on a `.promise()` to the end, then you can `.then` it